webpack/assets/javascripts/react_app/components/Pagination/Pagination.js

```diff
@@ -2,25 +2,26 @@ import React from 'react';
 import Proptypes from 'prop-types';
 import { isEmpty } from 'lodash';
 import { Paginator } from 'patternfly-react';
-import { getURI, getURIpage, getURIperPage, changeQuery, translatePagination } from './PaginationHelper';
+import { resetPage, updateOnChange, getURI, getURIpage, getURIperPage, changeQuery, translatePagination } from './PaginationHelper';
 import './pagination.scss';
 
 const Pagination = ({ data }) => {
   const urlPage = getURIpage();
-  const urlPerPage = getURIperPage();
+  const currPerPage = getURIperPage() || data.perPage;
   const className = isEmpty(data.classNames) ? 'col-md-12' : `col-md-12 ${data.classNames.pagination_classes}`;
+  const perPageSelectFnc = resetPage(updateOnChange(changeQuery, 'per_page', currPerPage));
```

Comment:
you have here composition of three functions, which decreases code flow. could you simplify it a bit?

Reply:
To merge `resetPage` and `updateOnChange` into one function?
Or to change `changeQuery` core functionality?

I would rather do the first choice, though.

Reply:
the first choice sounds good to me  :+1: